tests/unit/tuf_repository_service_worker/test_repository.py

```diff
@@ -3311,3 +3267,124 @@ def fake_get_fresh(key):
         assert test_repo.write_repository_settings.calls == [
             pretend.call("ROOT_SIGNING", "fake_metadata")
         ]
+
+    @pytest.mark.parametrize(
```

Comment:
We had a discussion with @kairoaraujo and @KAUTH regarding table testing done in this way and we are all with mixed feelings.
The biggest issues I see with them:
1. it's harder to debug which test case fails when you see only `test_sign_metadata__update` with a specific assert that not always point you to the correct use case.
2. it could be a little complicated for new users.

What do you think @kairoaraujo and @KAUTH ?

Reply:
> We had a discussion with @kairoaraujo and @KAUTH regarding table testing done in this way and we are all with mixed feelings. The biggest issues I see with them:

Yeah, I figured. That's why I used the copy/paste approach for the first two main test cases, and only refactored on top of that. (Note though that the table does have a few extra test cases).
> 
> 1. it's harder to debug which test case fails when you see only `test_sign_metadata__update` with a specific assert that not always point you to the correct use case.

That's not necessarily true, if you look at the [log](https://github.com/repository-service-tuf/repository-service-tuf-worker/actions/runs/5890277848/job/15975064496?pr=355#step:6:195) you'll see that each call of the method is listed as separat test.

> 2. it could be a little complicated for new users.

🤷 I guess, that's debatable. If we expect devs to get familiar with pytest, monkey patching, pretend, mock, etc., we might as well ask them to read pytest docs for `parametrize`.  I personally find thousands of lines of testing code more intimidating. But yeah I agree with the intention of making testing code as little engineered as possible.

Reply:
We need to decide about this test and methodology @kairoaraujo and @lukpueh.
Do we want to meet or discuss it here?

PS: Now when I read it it's not immediately clear to me which test case is testing what.
Maybe if there was some kind of a title of a test case would be better.
Something familiar to what we did in python-tuf here: https://github.com/theupdateframework/python-tuf/blob/038ecd65dc6b74e9d0306d3c57c3e7d621155e47/tests/test_updater_key_rotations.py#L92